aspnet/fundamentals/dependency-injection.rst

```diff
@@ -51,51 +50,47 @@ Registering Your Own Services
 
 .. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Controllers/CharactersController.cs
   :language: c#
-  :linenos:
-  :lines: 1-32
-  :emphasize-lines: 10,12,14,20,23-27
+  :lines: 8-36
+  :dedent: 4
+  :emphasize-lines: 3,5-8,14,21,23-26
 
 The `ICharacterRepository` simply defines the two methods the controller needs to work with `Character` instances.
 
 .. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Interfaces/ICharacterRepository.cs
   :language: c#
-  :linenos:
+  :emphasize-lines: 8-9
 
 This interface is in turn implemented by a concrete type, ``CharacterRepository``, that is used at runtime.
 
 .. note:: The way DI is used with the ``CharacterRepository`` class is a general model you can follow for all of your application services, not just in "repositories" or data access classes.
 
 .. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Models/CharacterRepository.cs
   :language: c#
-  :linenos:
+  :emphasize-lines: 9,11-14
 
-Note that ``CharacterRepository`` requests an ``ApplicationDbContext`` in its constructor. It is not unusual for dependency injection to be used in a chained fashion like this, with each requested dependency in turn requesting its own dependencies. The container is responsible for resolving all of the dependencies in the tree and returning the fully resolved service.
+Note that ``CharacterRepository`` requests an ``ApplicationDbContext`` in its constructor. It is not unusual for dependency injection to be used in a chained fashion like this, with each requested dependency in turn requesting its own dependencies. The container is responsible for resolving all of the dependencies in the graph and returning the fully resolved service.
 
 .. note:: Creating the requested object, and all of the objects it requires, and all of the objects those require, is sometimes referred to as an `object graph`. Likewise, the collective set of dependencies that must be resolved is typically referred to as a `dependency tree` or `dependency graph`.
 
-In this case, both ``ICharacterRepository`` and in turn ``ApplicationDbContext`` must be registered with the services container in ``ConfigureServices`` in ``Startup``. ``ApplicationDbContext`` is configured via the call to the extension method ``AddEntityFramework`` which includes an extension for adding a ``DbContext`` (``AddDbContext<T>``). Registration of the repository is done at the bottom end of ``ConfigureServices``:
+In this case, both ``ICharacterRepository`` and in turn ``ApplicationDbContext`` must be registered with the services container in ``ConfigureServices`` in ``Startup``. ``ApplicationDbContext`` is configured via the call to the extension method ``AddEntityFramework`` which includes an extension for adding a ``DbContext`` (``AddDbContext<T>``). Registration of the repository is done in ``ConfigureServices``:
```

Comment:
Replace  Registration of the repository is done in ``ConfigureServices``: 
with 
The following code shows the CharacterRepository registration (or something like that)
Then use:

.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Startup.cs
  :language: c#
  :lines: 37-55,63
  :dedent: 8
  :emphasize-lines: 19